### PR TITLE
Fix convex function ignoring given properties

### LINKED ISSUE #2762: turf.convex ignores given properties
# Background

In my sample code below, the function combines a list of adjacent polygons into one. Every polygon shows its cadastral information such as cadastral community number, as well as plot number. When combining these polygons into a single convex, I want to combine the numbers into list, that is populated in the convex additional properties. The function seems to ignore the given properties though. [Please observe the source code of the convex function](packages/turf-convex/index.ts).

I would like to open a pull request to fix this problem. I just need approval or confirmation from someone that they were able to reproduce the problem.

- [x] The version of Turf you are using, and any other relevant versions.
6.5.0
- [ ] GeoJSON data as a [gist file](https://gist.github.com/) or [geojson.io](http://geojson.io/) (filename extension must be `.geojson`).
Issue is rather simple, and doesn't need test data. Should be reproducible with any GeoJSON.
- [x] Snippet of source code or for complex examples use [jsfiddle](https://jsfiddle.net/).
```ts
type AdditionalProperties = {
  cadastralCommunity: string[];
  plots: string[];
};

type TurfFeature = turf.Feature<turf.Polygon | turf.MultiPolygon, AdditionalProperties>;

function combinePolygons(polygons: TurfFeature[]) {
  const combinedPolygons = polygons.reduce<TurfFeature>((previousPolygon, currentPolygon) => {
    if (previousPolygon == null) {
      return currentPolygon;
    }

    const reducedPlotNumbers = [...previousPolygon.properties.plots, ...currentPolygon.properties.plots];
    const reducedCadastralCommunities = [
      ...previousPolygon.properties.cadastralCommunity,
      ...currentPolygon.properties.cadastralCommunity,
    ];

    return turf.union<AdditionalProperties>(previousPolygon, currentPolygon, {
      properties: {
        plots: reducedPlotNumbers,
        cadastralCommunity: reducedCadastralCommunities,
      },
    });
  }, null);

  if (combinedPolygons.geometry.type === 'Polygon') {
    return combinedPolygons;
  }

  const { properties } = combinedPolygons;
  const combinedConvex = turf.convex<AdditionalProperties>(combinedPolygons.geometry, {
    // there's a bug in the convex function where the properties are not populated https://github.com/Turfjs/turf/blob/master/packages/turf-convex/index.ts
     properties: combinedPolygons.properties
  });

  if (combinedConvex) {
    // this should not be necessary
    combinedConvex.properties = properties;
  }

  return combinedConvex;
}
```
- [x] Verify this issue hasn't already been reported, or resolved in the latest alpha pre-release.
I couldn't find an issue matching this reported issue.

### PR DESCRIPTION
**Pull request by kareemjeiroudi**
# Background

Using the `convex` function to combine a feature collection returns a newly constructed polygon with missing properties. This is crucial if the caller of the function attempts to pass additional properties to the end polygon.

PR includes a test to test this scenario and improved typing

### Resolve #2762

- [x] Is this a bug fix, new functionality, or a breaking change?
This change fixes a bug where the convex function ignores given properties. I came across this problem, as I was working on a project for one of our clients 
- [x] Have read and followed the steps for [preparing a pull request](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#preparing-a-pull-request).

### CODE REVIEW

File: packages/turf-convex/test.ts [hunk 1]
```diff
@@ -32,3 +33,17 @@ test("turf-convex -- empty", (t) => {
   t.deepEqual(convex(featureCollection([])), null, "corner case: null hull");
   t.end();
 });
+
+test("turf-convex -- properties are returned", (t) => {
```
Comment (smallsaucepan) [inline packages/turf-convex/test.ts:37]: Could you please make the test name more descriptive? Maybe "properties are transferred to result polygon" instead?

**Review by smallsaucepan — CHANGES_REQUESTED** [review]
Thanks for this @kareemjeiroudi. Just a request for a minor change to test.ts and we should be good to go!

File: packages/turf-convex/test.ts (hunk 1 above)
Comment (kareemjeiroudi) [reply packages/turf-convex/test.ts:37]: Sure. Thanks for the feedback. I'll get back to this as soon as back from holidays.

Comment (kareemjeiroudi) [reply packages/turf-convex/test.ts]: @smallsaucepan I sincerely apologize for my late late late response. But it’s better late than never. 
I fixed the description of the test. Let me know if there’s anything else missing 👍

**Review by smallsaucepan — APPROVED** [review]

**Comment by kareemjeiroudi** [comment]
I had to rebase on master

**Comment by smallsaucepan** [comment]
Hi @kareemjeiroudi would like to merge this PR but can't bring it up to date with master first. Do you know if you have "Allow edits by maintainers" disabled on your branch / repo?

**Comment by blchelle** [comment]
Hi all, are there any plans to get this merged? I can submit another PR to resolve this if necessary.

**Comment by smallsaucepan** [comment]
That would actually be a huge help thanks @blchelle. I know it's not a massive PR, though could you please include both yourself and @kareemjeiroudi in the author list? That seems most fair.

**Comment by blchelle** [comment]
Thanks @smallsaucepan, I have put up the PR. Let me know if anything needs to be changed.
https://github.com/Turfjs/turf/pull/2972

**Comment by kareemjeiroudi** [comment]
Hi everyone, apologies for not actively updating this PR. Thanks @blchelle for taking over. @smallsaucepan I'll close this PR, as soon as the other PR is merged 👍.

**Comment by kareemjeiroudi** [comment]
> Hi @kareemjeiroudi would like to merge this PR but can't bring it up to date with master first. Do you know if you have "Allow edits by maintainers" disabled on your branch / repo?

@smallsaucepan  I've just rebased the branch on master and checked "Allow edits and access to secrets by maintainers"

**Comment by smallsaucepan** [comment]
Thanks @kareemjeiroudi. I reckon we'll go with #2972 as planned, just to avoid to-ing and fro-ing. Will hopefully take a look at that soon. Really do appreciate your help putting it together initially.

**Comment by mfedderly** [comment]
Closing out as this was fixed with #2972
